docs: update readmes (#3315)
digdir/designsystemet: README.md

</div>

Designsystemet is a suite of tools used to create design systems.

Our goal is to create consistent and user-friendly experiences in digital solutions for public services, making them more efficient and reliable.

## 🔗 Links

[Storybook](https://storybook.designsystemet.no/) - Preview for React components.

[Storefront](https://designsystemet.no/) - General documentation.

> [Theme](https://theme.designsystemet.no/) - Theme-builder.

Reviewer: ```suggestion
[Theme](https://theme.designsystemet.no/) - Themebuilder.
```

## 📦 Packages

[`@digdir/designsystemet`](https://www.npmjs.com/package/@digdir/designsystemet) - CLI for Designsystemet.

[`@digdir/designsystemet-css`](https://www.npmjs.com/package/@digdir/designsystemet-css) - CSS implementation of Designsystemet components.

> [`@digdir/designsystemet-react`](https://www.npmjs.com/package/@digdir/designsystemet-react) - React implementation of Designsystemet components.

Reviewer: ```suggestion
[`@digdir/designsystemet-react`](https://www.npmjs.com/package/@digdir/designsystemet-react) - Headless React implementation of Designsystemet components.
```

Author: They are not headless since they are linked to class names from `@digdir/designsystemet-css`.

Reviewer: They don't provide styling themselves though 🤔 
A separate package is needed

Author: Yeah, but just because you leave out they styling doesn't mean they are headless per se.

Reviewer: > a headless component is a component that provides functionality and behavior without dictating the rendered output. In other words, it's 'headless' because it does not dictate the 'look' of the component—the 'head' or visual rendering is up to you.

Would this not describe our components?

Author: No. We dictate their styling by saying e.g. `Input` should be styled by `ds-input`. Headless components have no internal defined classnames.

[`@digdir/designsystemet-theme`](https://www.npmjs.com/package/@digdir/designsystemet-theme) - Digdir themes for Designsystemet.

## 🚀 Get started

> Follow these steps to get started with the Designsystem in code.

Reviewer: ```suggestion
Follow these steps to get started with Designsystemet in code.
```
"the Designsystem" eller "Designsystemet"?

### 1. Install

> Install the minimum to get started with Designsystemet 

Reviewer: ```suggestion
Install the essential packages to get started with Designsystemet:
```

```sh
@digdir/designsystemet-css 
@digdir/designsystemet-theme # or custom theme 
```

Install `@digdir/designsystemet-theme` if you need any of the Digdir themes. Uses `digdir` theme by default.

#### 1.1 Custom theme

Create your own theme for Designsystemet by going to our [theme-builder](https://theme.designsystemet.no/).

Designsystemet theming is defined using [design-tokens](https://www.uxpin.com/studio/blog/what-are-design-tokens). 
This is done so that you can use [Token Studio](https://tokens.studio/) to sync your theme in code with [Designsystemet Figma UI kit](https://www.figma.com/community/file/1322138390374166141/designsystemet-core-ui-kit), in addition to provide future flexibility.

Run `npx @digdir/designsystemet tokens build` to build CSS files for your custom theme (from the design-tokens) which you can then import and use with Designsystemet.

Using a custom theme you can skip the `@digdir/designsystemet-theme`.

#### 1.2 React

Install `@digdir/designsystemet-react` if you want to use the React components.

```jsx

> import '@digdir/designsystemet-theme'; // or custom theme CSS file
> import '@digdir/designsystemet-css';

Reviewer: ```suggestion
import '@digdir/designsystemet-css';
import '@digdir/designsystemet-theme'; // or custom theme CSS file
```
For å være lik blokka over

    // ...other compilerOptions
    "types": [
      // ...other types
      "@digdir/designsystemet-theme" or "<custom-theme>/colors.d.ts",
      "@digdir/designsystemet-react/react-types",
      ]
  },
}
```

### 2. Font

You are free to use any font-family.

The components are designed and developed using the [Inter font](https://github.com/rsms/inter) so variations might occur if a different font is used.

#### 2.1 Use Inter font (optional)

Add the `<link>` tag in `<head>`, and set `font-family` to `Inter` in your global css file.

##### HTML

```html
<link
  rel="stylesheet"
  href="https://altinncdn.no/fonts/inter/v4/inter.css"
/>
```

##### CSS

```css
body {
  font-family: 'Inter', sans-serif;
  font-feature-settings: 'cv05' 1; /* Enable lowercase l with tail */
}
```

The `font-feature-settings` adds a tail to lowercase `L`'s.

> If you choose to install the font in a different way, remember to include the `400`, `500` and `600` font weights.

Reviewer: ```suggestion
If you chose to install the font in a different way, remember to include the `400`, `500` and `600` font weights.
```

Author: We are talking about present-tense. Chose is past-tense

Reviewer: Denne er forsåvidt rett begge vega - men når eg las teksten over, og så denne - så virka "chose" (valgte) bedre enn "choose" (velger)

Author: We are reminding them about something they need to do in the future when they add a font. "choose" is the correct tense.

Reviewer: ```suggestion
If you install the font in a different way, remember to include the `400`, `500` and `600` font weights.
```

## 🫶 Contributing

Learn how you can contribute to this project by reading our [Code of Conduct](./CODE_OF_CONDUCT.md) and [Contributing Guide](./CONTRIBUTING.md).

## 💪 Contributors

> We are lucky to have a great group of people who help out with Designsystemet

Reviewer: ```suggestion
We are lucky to have a great group of people who have contributed to Designsystemet
```
